logic/generateLogic.gradle

```
@@ -123,16 +123,19 @@ def createGenerateLogicTaskForPackage(
             def identifier = /[a-zA-Z0-9]+/
             def newLine = /(?:\r\n?|\n)/
             def newLineAndIndent = /$newLine\s*/
-            def parameter = /,(?: |$newLineAndIndent)($identifier): ([^:]+?)/
-            def returnType = /(?:$newLineAndIndent)?\)(:.+)/
-
-            def typeIdentifier = / *fun (<.+?> )?($identifier)\((?:$newLineAndIndent)?$identifier: ($identifier(?:\.$identifier)*<.+>)/
-            def patterns = (6..0).collect {
-                def steps = (0..<it * 2).findAll { it % 2 == 0 }
+            def parameter = { paramNumber -> /,(?: |$newLineAndIndent)(?<paramName$paramNumber>$identifier): (?<typeName$paramNumber>[^:]+?)/}
+            def returnType = /(?:$newLineAndIndent)?\)(?<returnType>:.+)/
+
+            def typeIdentifier = / *fun (?<generics><.+?> )?(?<funcName>$identifier)\((?:$newLineAndIndent)?$identifier: (?<type>$identifier(?:\.$identifier)*<.+>)/
+            def patterns = (6..0).collect { int amountOfParameters ->
+                def params = []
+                if (amountOfParameters > 0) {
+                    params = (1..amountOfParameters).toList()
```

tiny detail, toList is not required
```suggestion
                    params = (1..amountOfParameters)
```